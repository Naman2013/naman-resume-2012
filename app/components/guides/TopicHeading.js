import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './TopicHeading.style';

const TopicHeading = ({ text }) => (
  <Fragment>
    <h2>{text}</h2>
    <style jsx>{style}</style>
  </Fragment>
);

TopicHeading.propTypes = {
  text: PropTypes.string.isRequired,
};

export default TopicHeading;
