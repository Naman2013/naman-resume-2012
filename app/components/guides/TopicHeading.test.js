import React from 'react';
import { shallow } from 'enzyme';
import TopicHeading from './TopicHeading';

describe('TopicHeading', () => {
  const shallowWrapper = shallow(<TopicHeading text="Foo" />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
