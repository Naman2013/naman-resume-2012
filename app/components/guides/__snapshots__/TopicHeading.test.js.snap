// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicHeading should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopicHeading
    text="Foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h2
          className="jsx-2575439962"
        >
          Foo
        </h2>,
        <JSXStyle
          css={
            Array [
              "h2.jsx-2575439962{font-family:adobe-garamond-pro, serif;color:#415671;font-size:22px;border-bottom:1px solid #d9dee5;margin:0 35px;padding-top:40px;padding-bottom:20px;text-transform:capitalize;}",
              "@media screen and (min-width: 768px){h2.jsx-2575439962{font-size:32px;border-bottom:3px solid #415671;margin:0;}}",
            ]
          }
          styleId="2575439962"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Foo",
          "className": "jsx-2575439962",
        },
        "ref": null,
        "rendered": "Foo",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "h2.jsx-2575439962{font-family:adobe-garamond-pro, serif;color:#415671;font-size:22px;border-bottom:1px solid #d9dee5;margin:0 35px;padding-top:40px;padding-bottom:20px;text-transform:capitalize;}",
            "@media screen and (min-width: 768px){h2.jsx-2575439962{font-size:32px;border-bottom:3px solid #415671;margin:0;}}",
          ],
          "styleId": "2575439962",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h2
            className="jsx-2575439962"
          >
            Foo
          </h2>,
          <JSXStyle
            css={
              Array [
                "h2.jsx-2575439962{font-family:adobe-garamond-pro, serif;color:#415671;font-size:22px;border-bottom:1px solid #d9dee5;margin:0 35px;padding-top:40px;padding-bottom:20px;text-transform:capitalize;}",
                "@media screen and (min-width: 768px){h2.jsx-2575439962{font-size:32px;border-bottom:3px solid #415671;margin:0;}}",
              ]
            }
            styleId="2575439962"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foo",
            "className": "jsx-2575439962",
          },
          "ref": null,
          "rendered": "Foo",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "h2.jsx-2575439962{font-family:adobe-garamond-pro, serif;color:#415671;font-size:22px;border-bottom:1px solid #d9dee5;margin:0 35px;padding-top:40px;padding-bottom:20px;text-transform:capitalize;}",
              "@media screen and (min-width: 768px){h2.jsx-2575439962{font-size:32px;border-bottom:3px solid #415671;margin:0;}}",
            ],
            "styleId": "2575439962",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
